### PR TITLE
Unified `PlotAccessor` for DataFrame and Series

### LINKED ISSUE #1661: Simplify the Plot Assessors
Current `plot` accessors have separate implementations for both `Series` and `DataFrame` despite having a very similar API structure.  Since there exist separate functions for both `plot_series` and `plot_frame`, a single class can handle plots for both.

I will open a PR for to solve this issue, ([cross ref](https://github.com/databricks/koalas/pull/1639#issuecomment-659824221))

### PR DESCRIPTION
**Pull request by DumbMachine**
- [x] Ready for review

Aims to fix #1661
New `PlotAccessor` works the same way as the old.

## TODO:
- [ ] Confirm each function has appropriate example and options in the `docs`

Examples:
```python
import pandas as pd
import databricks.koalas as ks
import matplotlib.pyplot as plt
kf =  ks.DataFrame([[1, 2, 3, 4], [5, 6, 7, 8]], columns=["A", "B", "C", "D"])
```
```python
# dataframe like
print(kf.plot)
kf.plot(kind="line")
kf.plot(kind="line", backend="plotly", width=400, height=400).show()
```
`<databricks.koalas.plot.PlotAccessor object at 0x7f32d669b590>`
image
image

```python
# series like
print(kf.A.plot)
kf.A.plot(kind="line");
kf.A.plot(kind="line", backend="plotly", width=400, height=400).show()
```
`<databricks.koalas.plot.PlotAccessor object at 0x7f32cdffd750>`
image
image

### Error are handled, same as before
```python
kf.plot(kind="pie", legend=False)
kf.plot(kind="pie",  backend="plotly", width=400, height=400).show()
```
image
```bash
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-13-9597ff26ce78> in <module>
      1 kf.plot(kind="pie", legend=False)
----> 2 kf.plot(kind="pie",  backend="plotly", width=400, height=400).show()

~/work/koalas/databricks/koalas/plot.py in __call__(self, kind, backend, **kwargs)
   1322 
   1323         if plot_backend.__name__ != "databricks.koalas.plot":
-> 1324             return plot_backend.plot(plot_data, kind=kind, **kwds)
   1325 
   1326         if kind not in self._all_kinds:

~/anaconda3/lib/python3.7/site-packages/plotly/__init__.py in plot(data_frame, kind, **kwargs)
    101         return histogram(data_frame, **new_kwargs)
    102     raise NotImplementedError(
--> 103         "kind='%s' not yet supported for plotting.backend='plotly'" % kind
    104     )
    105 

NotImplementedError: kind='pie' not yet supported for plotting.backend='plotly'
```
And finally
```python
# exists for dataframe like
kf.plot(kind="scatter", x='A', y='B')
# does not exist for series like
kf.A.plot(kind="scatter")
```
image
```bash
---------------------------------------------------------------------------
PandasNotImplementedError                 Traceback (most recent call last)
<ipython-input-6-674f15f3badc> in <module>
----> 1 kf.A.new_plot(kind="scatter")

~/work/koalas/databricks/koalas/new_plot.py in __call__(self, kind, backend, **kwargs)
    157         if isinstance(self.data, Series):
    158             if kind not in self._series_kinds:
--> 159                 return  unsupported_function(class_name="pd.Series", method_name=kind)()
    160             return plot_series(data=self.data, kind=kind, **kwds)
    161         elif isinstance(self.data, DataFrame):

~/work/koalas/databricks/koalas/missing/__init__.py in unsupported_function(*args, **kwargs)
     21     def unsupported_function(*args, **kwargs):
     22         raise PandasNotImplementedError(
---> 23             class_name=class_name, method_name=method_name, reason=reason
     24         )
     25 

PandasNotImplementedError: The method `pd.Series.scatter()` is not implemented yet.
```

### CODE REVIEW

**Comment by DumbMachine** [comment]
This happened in the last PR as well. Running : 
```
(base) ➜  koalas git:(master) ✗ dev/reformat 
```
Seems to make incorrect changes.
```python
starting python compilation test...
Python compilation failed with the following errors:
*** Error compiling './build/lib/databricks/koalas/plot.py'...
  File "./build/lib/databricks/koalas/plot.py", line 386
    **kwargs,
            ^
SyntaxError: invalid syntax

*** Error compiling './databricks/koalas/plot.py'...
  File "./databricks/koalas/plot.py", line 386
    **kwargs,
            ^
SyntaxError: invalid syntax
```

**Comment by DumbMachine** [comment]
Could anybody help me with the PIP error? I do not get any error during local testing.

**Comment by ueshin** [comment]
I submitted a PR to you branch https://github.com/DumbMachine/koalas/pull/2.

File: databricks/koalas/plot.py [hunk 1]
```diff
@@ -1186,181 +1185,170 @@ def _get_plot_backend(backend=None):
...
+                return unsupported_function(class_name="pd.Series", method_name=kind)()
+            return plot_series(data=self.data, kind=kind, **kwds)
+        elif isinstance(self.data, DataFrame):
+            if kind not in self._dataframe_kinds:
+                return unsupported_function(class_name="pd.DataFrame", method_name=kind)()
+            return plot_frame(data=self.data, kind=kind, **kwds)
+
+    # added this, as it was present before
+    __call__.__doc__ = plot_frame.__doc__
     __call__.__doc__ = plot_series.__doc__
```
Comment (ueshin) [inline databricks/koalas/plot.py:1348]: This is just overwriting the exiting doc.

**Comment by ueshin** [comment]
@DumbMachine Could you push an empty commit to run tests again?
Seems like the current build doesn't include the new changes for some reason.

File: databricks/koalas/plot.py (hunk 1 above)
Comment (DumbMachine) [reply databricks/koalas/plot.py]: I did not understand its use initially. Now I do. Instead of adding any particular functions documentation should I add doc string to `__call__` method?

File: databricks/koalas/plot.py [hunk 2]
```diff
@@ -1186,181 +1186,168 @@ def _get_plot_backend(backend=None):
...
     ``s.plot(kind='hist')`` is equivalent to ``s.plot.hist()``
     """
 
+    # from databricks.koalas import DataFrame, Series
+
+    _common_kinds = ("area", "bar", "barh", "box", "hist", "kde", "line", "pie")
+    _series_kinds = () + _common_kinds
+    _dataframe_kinds = ("scatter", "hexbin") + _common_kinds
+    _kind_aliases = {"density": "kde"}
+    _all_kinds = _common_kinds + _series_kinds + _dataframe_kinds
```
Comment (ueshin) [inline databricks/koalas/plot.py:1207]: The name `_all_kinds` could be ambiguous since it exists in this file.
Could you rename it?

Comment (ueshin) [inline databricks/koalas/plot.py:1207]: Btw, these seems ok to be `set`?

**Comment by ueshin** [comment]
@DumbMachine Could you address my comments above?

File: databricks/koalas/plot.py [hunk 3]
```diff
@@ -1186,181 +1186,168 @@ def _get_plot_backend(backend=None):
...
+        if isinstance(self.data, Series):
+            if kind not in self._series_kinds:
+                return unsupported_function(class_name="pd.Series", method_name=kind)()
+            return plot_series(data=self.data, kind=kind, **kwds)
+        elif isinstance(self.data, DataFrame):
+            if kind not in self._dataframe_kinds:
+                return unsupported_function(class_name="pd.DataFrame", method_name=kind)()
+            return plot_frame(data=self.data, kind=kind, **kwds)
+
+    __call__.__doc__ = plot_frame.__doc__
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply databricks/koalas/plot.py:1346]: Haven't built it and checked by myself but I think it's okay to consolidate in general.

**Comment by DumbMachine** [comment]
Sorry for the delay, have made the changes. I will modify if any further changes are required,

File: databricks/koalas/plot.py [hunk 4]
```diff
@@ -1684,20 +1692,12 @@ def hist(self, bins=10, **kwds):
...
             >>> ax = s.plot.hist()
-
-        For DataFrame:
-        .. plot::
-            :context: close-figs
-
-            >>> df = ks.DataFrame({'mass': [0.330, 4.87, 5.97],
-            ...                    'radius': [2439.7, 6051.8, 6378.1]},
-            ...                   index=['Mercury', 'Venus', 'Earth'])
-            >>> plot = df.plot.hist(figsize=(6, 3))
```
Comment (ueshin) [inline databricks/koalas/plot.py]: Why removed the example?

Comment (ueshin) [reply databricks/koalas/plot.py]: Btw, where is this example from? Seems like the original example for hist was:

```py
            >>> df = pd.DataFrame(	
            ...     np.random.randint(1, 7, 6000),	
            ...     columns=['one'])	
            >>> df['two'] = df['one'] + np.random.randint(1, 7, 6000)	
            >>> df = ks.from_pandas(df)	
            >>> ax = df.plot.hist(bins=12, alpha=0.5)	
```


https://github.com/databricks/koalas/pull/1662/files#diff-68e0bb2c8c2742f7ee4d267dbf3d1c68L2143-L2148

File: databricks/koalas/plot.py [hunk 5]
```diff
@@ -1344,6 +1344,7 @@ def __call__(self, kind="line", backend="matplotlib", **kwargs):
             return plot_frame(data=self.data, kind=kind, **kwds)
 
     __call__.__doc__ = plot_frame.__doc__
+    __call__.__doc__ = plot_series.__doc__
```
Comment (ueshin) [inline databricks/koalas/plot.py:1347]: Why is this added back?

Comment (ueshin) [reply databricks/koalas/plot.py:1347]: btw, I guess we can remove both for now since it doesn't appear in the doc.

**Comment by ueshin** [comment]
Thanks! merging.
